src/main/webapp/app/quiz/re-evaluate/multiple-choice-question/re-evaluate-multiple-choice-question.component.js

```diff
@@ -0,0 +1,378 @@
+            (vm.question.explanation ? "\n\t[-e] " + vm.question.explanation : "")
+        );
+        editor.setValue(markdownText);
+        editor.clearSelection();
+    }
+
+    /**
+     * generate the markdown text for this question
+     *
+     * The markdown is generated according to these rules:
+     *
+     * 1. First the answer text is inserted
+     * 2. If hint and/or explanation exist, they are added after the text with a linebreak and tab in front of them
+     *
+     * @param answer {answerOption}  is the AnswerOption, which the Markdown-field presents
+     */
+    function generateAnswerMarkdown(answer) {
+        var answerEditor = ace.edit("answer-content-editor-" + answer.id);
+        var markdownText = (
+            (answer.isCorrect ? "[x]" : "[ ]") + " " +
+            answer.text +
+            (answer.hint ? "\n\t[-h] " + answer.hint : "") +
+            (answer.explanation ? "\n\t[-e] " + answer.explanation : "")
+        );
+        answerEditor.setValue(markdownText);
+        answerEditor.clearSelection();
+    }
+
+    /**
+     * Parse the question-markdown and apply the result to the question's data
+     *
+     * The markdown rules are as follows:
+     *
+     * 1. Text is split at [x] and [ ] (also accepts [X] and [])
+     *    => The first part (any text before the first [x] or [ ]) is the question text
+     * 2. The question text is split at [-h] and [-e] tags.
+     *    => First part is text. Everything after [-h] is Hint, anything after [-e] is explanation
+     *
+     * @param text {string} the markdown text to parse
+     */
+    function parseQuestionMarkdown(text) {
+        // first split by [], [ ], [x] and [X]
+        var questionParts = text.split(/\[\]|\[ \]|\[x\]|\[X\]/g);
+        var questionText = questionParts[0];
+
+        // split question into main text, hint and explanation
+        var questionTextParts = questionText.split(/\[\-e\]|\[\-h\]/g);
+        vm.question.text = questionTextParts[0].trim();
+        if (questionText.indexOf("[-h]") !== -1 && questionText.indexOf("[-e]") !== -1) {
+            if (questionText.indexOf("[-h]") < questionText.indexOf("[-e]")) {
+                vm.question.hint = questionTextParts[1].trim();
+                vm.question.explanation = questionTextParts[2].trim();
+            } else {
+                vm.question.hint = questionTextParts[2].trim();
+                vm.question.explanation = questionTextParts[1].trim();
+            }
+        } else if (questionText.indexOf("[-h]") !== -1) {
+            vm.question.hint = questionTextParts[1].trim();
+            vm.question.explanation = null;
+        } else if (questionText.indexOf("[-e]") !== -1) {
+            vm.question.hint = null;
+            vm.question.explanation = questionTextParts[1].trim();
+        } else {
+            vm.question.hint = null;
+            vm.question.explanation = null;
+        }
+    }
```

Comment:
This code is duplicated from edit-multiple-choice-question.component.js. This is REALLY REALLY BAD!!!
Please talk to Valentin and make sure that this code is not duplicated, but that the function is called in both files:
- edit-multiple-choice-question.component.js
- re-evaluate-multiple-choice-question.component.js